### PR TITLE
Galois

### PR DESCRIPTION
**Pull request by martyall**
Use `galois-field` library `Prime` field type instead of `Rational` everywhere

### CODE REVIEW

**Comment by martyall** [comment]
This change mostly consists of

- Replacing the use of Rational with `Data.Field.Galois.Prime` everywhere (technically something like `KnownNat p => Prime p`
- Replacing the custom `Field` typeclass with `Data.Field.Galois.PrimeField`
- replacing the use of `Field` methods (e.g. `add`, `mult`) with the normal haskell `Num` operations e.g. `(+)`, (*)`
- replacing the use of `Field` constants (e.g. `zero`, `one`) with normal values `0`, `1`.

**Comment by martyall** [comment]
@kejace Can I get you to go through these changes and make sure that the arithmetic parts transferred over correctly,

e.g.

`add` |-> `(+)`
`zero` |-> `0`
`negate a` |-> `-a`
etc?

I actually found one mistake already which didn't fail the tests and would have been really hard to track down later.

File: src/Snarkl/Interp.hs [hunk 1]
```diff
@@ -127,31 +127,28 @@ interpExpr e = case e of
...
         op a b = case _op of
-          Add -> pure $ a `add` b
-          Sub -> pure $ a `add` neg b
-          Mult -> pure $ a `mult` b
-          Div -> pure $
-            case inv b of
-              Nothing -> failWith $ ErrMsg $ show b ++ " not invertible"
-              Just b' -> a `mult` b'
+          Add -> pure $ a + b
+          Sub -> pure $ a - b
```
Comment (kejace) [inline src/Snarkl/Interp.hs:143]: Possibly worth doing `pure $ a + neg b` here, though I don't see how it would make a difference.

**Review by kejace — APPROVED** [review]
I went through with a fine-toothed comb 🕵️

**Pull request merged**